backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/incremental/provider/NodePositionProvider.java

```diff
@@ -51,116 +54,160 @@
-            if (maxBottom == null) {
-                newPosition = this.getNewChildPosition(node);
+    private Position getPositionRelativeToParent(NodeLayoutData node, SinglePositionEvent diagramElementEvent) {
+        Position position;
+        IContainerLayoutData parent = node.getParent();
+        Position eventPosition = diagramElementEvent.getPosition();
+        if (Position.UNDEFINED.equals(parent.getPosition())) {
+            position = this.getDefaultPosition(node);
+        } else {
+            if (parent instanceof DiagramLayoutData) {
+                position = eventPosition;
+            } else if (this.isEventPositionInNodeBounds(parent, eventPosition)) {
+                Position parentPosition = parent.getPosition();
+                double xPosition = eventPosition.getX() - parentPosition.getX();
+                double yPosition = eventPosition.getY() - parentPosition.getY();
+                position = Position.at(xPosition, yPosition);
             } else {
-                double newY = maxBottom + this.getNodeGap(node);
-                newPosition = Position.at(0, newY);
+                position = this.getDefaultPosition(node);
             }
-        } else {
-            Position lastPosition = this.last.getPosition();
-            Size lastSize = this.last.getSize();
-            double x = lastPosition.getX();
-            double y = lastPosition.getY() + lastSize.getHeight() + this.getNodeGap(node);
-            newPosition = Position.at(x, y);
         }
-        this.last = node;
-        return newPosition;
+        return position;
     }
 
-    private Optional<Position> getSpecificNodePositionFromEvent(Optional<IDiagramEvent> optionalDiagramElementEvent, NodeLayoutData node) {
-        Position position = null;
-        if (optionalDiagramElementEvent.isPresent()) {
-            IDiagramEvent diagramElementEvent = optionalDiagramElementEvent.get();
-            if (diagramElementEvent instanceof MoveEvent) {
-                position = this.getPossiblePositionFromMoveEvent(node, (MoveEvent) diagramElementEvent);
-            } else if (diagramElementEvent instanceof ResizeEvent) {
-                position = this.getPossiblePositionFromResizeEvent(node, (ResizeEvent) diagramElementEvent);
-            }
+    private Optional<Position> getPositionRelativeToSibling(NodeLayoutData node) {
+        Optional<NodeLayoutData> lastPositionedSiblingOptional = this.getLastPositionedSibling(node);
+        if (lastPositionedSiblingOptional.isPresent()) {
+            NodeLayoutData lastPositionedSibling = lastPositionedSiblingOptional.get();
+            Position lastPosition = lastPositionedSibling.getPosition();
+            Size lastSize = lastPositionedSibling.getSize();
+            double x = lastPosition.getX();
+            double y = lastPosition.getY() + lastSize.getHeight() + IncrementalLayoutEngine.NODES_GAP;
+            return Optional.of(Position.at(x, y));
         }
```

Comment:
This is what is replacing the "last" field. It will not be as performant as before but it works with the siblings. Does it seems ok with you?